server/service/schedule/schedule.go

```diff
@@ -139,126 +153,221 @@ func New(
 //
 // All jobs must be added before calling Start.
 func (s *Schedule) Start() {
-	var m sync.Mutex // protects currentStart and currentWait.
-	currentStart := time.Now()
-	currentWait := 10 * time.Second
+	var intervalStartedAt time.Time // start time of the most recent run of the scheduled jobs
+	var m sync.Mutex                // protects intervalStartedAt
 
-	getWaitTimes := func() (start time.Time, wait time.Duration) {
+	getIntervalStartedAt := func() (start time.Time) {
 		m.Lock()
 		defer m.Unlock()
 
-		return currentStart, currentWait
+		return intervalStartedAt
 	}
 
-	setWaitTimes := func(start time.Time, wait time.Duration) {
+	setIntervalStartedAt := func(start time.Time) {
 		m.Lock()
 		defer m.Unlock()
 
-		currentStart = start
-		currentWait = wait
+		intervalStartedAt = start.Truncate(time.Second)
 	}
```

Comment:
I wonder if all of this, the interval, mutex, func, should now be moved to be a part of Schedule?

Reply:
Definitely considered that too. I wasn't sure if it made sense to move it out since there isn't a need to use any of those things outside of the `Start` method.  But happy to do so if you think it is a better pattern.

Reply:
I suppose it's fine, part of my reasoning was the size of `Start` itself. But up to you, not feeling super strongly about it.